Linq2Rest/Parser/FilterExpressionFactory.cs

```diff
@@ -177,11 +177,11 @@ private static Expression GetRightOperation(string token, Expression right)
             switch (function.ToLowerInvariant())
             {
                 case "substringof":
-                    return Expression.AndAlso(Expression.NotEqual(left, Expression.Constant(null, typeof(object))), Expression.Call(right, MethodProvider.ContainsMethod, new[] { left }));
+                    return Expression.AndAlso(Expression.NotEqual(left, Expression.Constant(null, typeof(object))), Expression.Call(Expression.Call(right, MethodProvider.ToLowerMethod), MethodProvider.ContainsMethod, new[] { left }));

```

Comment:
@AshwaniNagarro in order for this to work wouldn't we need to apply the ToLowerMethod to both the left and right?

Reply:
yes, you are right, I have added that